From 79af8c6f7123a20073ce38afd2c49e1d492a9cb7 Mon Sep 17 00:00:00 2001
From: DeadSix <DeadSix27@users.noreply.github.com>
Date: Wed, 6 Sep 2023 12:56:52 +0200
Subject: [PATCH 1/2] add pc file for windows

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c69b135..80690a1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -136,6 +136,10 @@ add_compile_definitions(
     $<$<CONFIG:>:NDEBUG>
 )
 
+set(prefix "${CMAKE_INSTALL_PREFIX}")
+set(exec_prefix "${CMAKE_INSTALL_PREFIX}")
+set(libdir "${CMAKE_INSTALL_PREFIX}/lib")
+set(includedir "${CMAKE_INSTALL_PREFIX}/include")
 
 ######################
 # System definitions #
@@ -1335,3 +1339,6 @@ if(BUILD_TESTING)
         )
     endforeach()
 endif()
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/liblzma/liblzma.pc.in ${CMAKE_CURRENT_BINARY_DIR}/liblzma.pc @ONLY)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/liblzma.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
-- 
2.42.0

