From 9f99074c9f4d7ee0c02371f7b2d0dec146764fae Mon Sep 17 00:00:00 2001
From: DeadSix <DeadSix27@users.noreply.github.com>
Date: Wed, 6 Sep 2023 12:57:11 +0200
Subject: [PATCH 2/2] temp llvm build fix

---
 CMakeLists.txt | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80690a1..1e2a5d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1019,13 +1019,13 @@ if(HAVE_GETOPT_LONG AND HAVE_DECODERS)
 
     target_link_libraries(xzdec PRIVATE liblzma)
 
-    if(WIN32)
+    # if(WIN32)
         # Add the Windows resource file for xzdec.exe.
-        target_sources(xzdec PRIVATE src/xzdec/xzdec_w32res.rc)
-        set_target_properties(xzdec PROPERTIES
-            LINK_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/common/common_w32res.rc"
-        )
-    endif()
+        # target_sources(xzdec PRIVATE src/xzdec/xzdec_w32res.rc)
+        # set_target_properties(xzdec PROPERTIES
+            # LINK_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/common/common_w32res.rc"
+        # )
+    # endif()
 
     tuklib_progname(xzdec)
 
@@ -1103,13 +1103,13 @@ if(NOT MSVC AND HAVE_GETOPT_LONG)
 
     target_compile_definitions(xz PRIVATE ASSUME_RAM=128)
 
-    if(WIN32)
+    # if(WIN32)
         # Add the Windows resource file for xz.exe.
-        target_sources(xz PRIVATE src/xz/xz_w32res.rc)
-        set_target_properties(xz PROPERTIES
-            LINK_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/common/common_w32res.rc"
-        )
-    endif()
+        # target_sources(xz PRIVATE src/xz/xz_w32res.rc)
+        # set_target_properties(xz PROPERTIES
+            # LINK_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/common/common_w32res.rc"
+        # )
+    # endif()
 
     tuklib_progname(xz)
     tuklib_mbstr(xz)
-- 
2.42.0

